### PR TITLE
Implement volume encryption for etcd and root amis

### PR DESCRIPTION
**Pull request by iuriaranda**
This is a PR to encrypt the k8s base image root volume and to encrypt etcd volumes as per https://github.com/skyscrapers/engineering/issues/55

### CODE REVIEW

File: cluster/main.tf [hunk 1]
```diff
@@ -13,16 +13,35 @@ data "aws_availability_zones" "available" {
...
+resource "aws_ami_copy" "k8s_base_image" {
+  count = "${var.k8s_image_encryption ? 1 : 0}"
+
+  # For now, this resource will replace the copied AMI everytime there's a new source AMI, so there'll always be just one AMI in the target account. Until this is implemented: https://github.com/terraform-providers/terraform-provider-aws/issues/792
+
+  name              = "${data.aws_ami.kubernetes_ami.name}"
+  description       = "Kubernetes base image"
+  source_ami_id     = "${data.aws_ami.kubernetes_ami.id}"
+  source_ami_region = "${data.aws_region.current.name}"
+  encrypted         = true
```
Comment (iuriaranda) [inline cluster/main.tf]: I think we should also set the kms key id (if set) for the encryption of the base ami

File: cluster/variables.tf [hunk 2]
```diff
@@ -145,3 +145,23 @@ variable "dns_provider" {
...
+  description = "Enable k8s image encryption"
+  default     = false
+}
+
+variable "etcd_encrypted_volumes" {
+  description = "Enable etcd volume encryption"
+  default     = "false"
+}
+
+variable "etcd_encryption_kms_key" {
```
Comment (iuriaranda) [inline cluster/variables.tf]: as with the storage class, you need the full key ARN for kops, so I would make it explicit in the name (`etcd_encryption_kms_key`) and in the description.

Btw, why not use the same key for both etcd and the root volume?

Comment (venturel) [reply cluster/variables.tf]: I think it gives more configuration freedom.

**Comment by duboisph** [comment]
@venturel For clarificatio, this only adds the option to enable etcd encryption. A migration path for existing clusters has not been tested yet?

**Comment by venturel** [comment]
@duboisph correct

**Pull request merged**